File: src/org/kobjects/nlp/api/Strings.java
package org.kobjects.nlp.api;

public class Strings {

  public static String[] splitAndTrim(String s, String deli) {
    String[] result = s.split(deli, -1);
    for (int i = 0; i < result.length; i++) {
      result[i] = result[i].trim();
    }
    return result;
  }

  public static String fill(String s, int len) {
     StringBuilder sb = new StringBuilder(len);
     sb.append(s);
     while (sb.length() < len) {
       sb.append(' ');
     }
     return sb.toString();
  }

  public static String supertrim(String s) {
  	StringBuilder sb = new StringBuilder(s.length());
  	boolean wasSpace = true;
  	for (int i = 0; i < s.length(); i++) {
  		char c = s.charAt(i);
  		if (c <= ' ') {
  			if (!wasSpace) {
  				sb.append(' ');
  				wasSpace = true;
  			} 
  		} else {
  			sb.append(c);
  			wasSpace = false;
  		}
  	}
  	if (wasSpace && sb.length() > 0) {
  		sb.setLength(sb.length() - 1);
  	}		
  	return sb.toString();
  }

}

File: src/api/Strings.js
export function splitAndTrim(str, deliminator) {
	const result = str.split(deliminator, -1);
	for (let i = 0; i < result.length; i++) {
		result[i] = result[i].trim();
	}
	return result;
}

export function fill(str, length) {
	for (let i = 0; i < length - str.length; i++) {
		str += ' ';
	}
	return str;
}

export function supertrim(str) {
	let wasSpace = true;
	for (let i = 0; i < str.length; i++) {
		const c = str.charAt(i);
		if (c <= ' ') {
			if (!wasSpace) {
				str += ' ';
				wasSpace = true;
			} 
		} else {
			str += c;
			wasSpace = false;
		}
	}
	if (wasSpace && str.length > 0) {
		str = str.slice(0, str.length - 1);
	}		
	return str;
}
